From: Debian Rust Maintainers <pkg-rust-maintainers@alioth-lists.debian.net>
Date: Sat, 2 Oct 2021 01:07:59 +0100
Subject: d-0000-ignore-removed-submodules

Description: remove upstream parts that are not needed for the Debian build, in
order to both reduce the orig tarball and the vendored crates within.

Forwarded: not-needed
---
 Cargo.toml                                          |  5 -----
 src/bootstrap/bootstrap.py                          |  4 ----
 src/bootstrap/src/core/build_steps/vendor.rs        |  5 -----
 src/bootstrap/src/core/builder/cargo.rs             |  5 +----
 src/bootstrap/src/core/builder/mod.rs               | 21 ---------------------
 src/build_helper/src/util.rs                        |  2 ++
 .../crates/rust-analyzer/src/config.rs              |  3 ++-
 7 files changed, 5 insertions(+), 40 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 67c7a9d..e273c4b 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -9,7 +9,6 @@ members = [
   "src/rustc-std-workspace/rustc-std-workspace-std",
   "src/rustdoc-json-types",
   "src/tools/build-manifest",
-  "src/tools/bump-stage0",
   "src/tools/cargotest",
   "src/tools/clippy",
   "src/tools/clippy/clippy_dev",
@@ -26,12 +25,8 @@ members = [
   "src/tools/lint-docs",
   "src/tools/lld-wrapper",
   "src/tools/llvm-bitcode-linker",
-  "src/tools/miri",
-  "src/tools/miri/cargo-miri",
   "src/tools/miropt-test-tools",
   "src/tools/opt-dist",
-  "src/tools/remote-test-client",
-  "src/tools/remote-test-server",
   "src/tools/replace-version-placeholder",
   "src/tools/run-make-support",
   "src/tools/rust-installer",
diff --git a/src/bootstrap/bootstrap.py b/src/bootstrap/bootstrap.py
index db7c7b6..89f04dc 100644
--- a/src/bootstrap/bootstrap.py
+++ b/src/bootstrap/bootstrap.py
@@ -1157,10 +1157,6 @@ class RustBuild(object):
         if "BOOTSTRAP_TRACING" in env:
             args.append("--features=tracing")
 
-        if self.use_locked_deps:
-            args.append("--locked")
-        if self.use_vendored_sources:
-            args.append("--frozen")
         if self.get_toml("metrics", "build"):
             args.append("--features")
             args.append("build-metrics")
diff --git a/src/bootstrap/src/core/build_steps/vendor.rs b/src/bootstrap/src/core/build_steps/vendor.rs
index 36a740c..4684c2d 100644
--- a/src/bootstrap/src/core/build_steps/vendor.rs
+++ b/src/bootstrap/src/core/build_steps/vendor.rs
@@ -21,14 +21,9 @@ pub fn default_paths_to_vendor(builder: &Builder<'_>) -> Vec<(PathBuf, Vec<&'sta
         ("src/tools/cargo/Cargo.toml", vec!["src/tools/cargo"]),
         ("src/tools/clippy/clippy_test_deps/Cargo.toml", vec![]),
         ("src/tools/rust-analyzer/Cargo.toml", vec![]),
-        ("compiler/rustc_codegen_cranelift/Cargo.toml", vec![]),
-        ("compiler/rustc_codegen_gcc/Cargo.toml", vec![]),
         ("library/Cargo.toml", vec![]),
         ("src/bootstrap/Cargo.toml", vec![]),
         ("src/tools/rustbook/Cargo.toml", SUBMODULES_FOR_RUSTBOOK.into()),
-        ("src/tools/rustc-perf/Cargo.toml", vec!["src/tools/rustc-perf"]),
-        ("src/tools/opt-dist/Cargo.toml", vec![]),
-        ("src/doc/book/packages/trpl/Cargo.toml", vec![]),
     ]
     .into_iter()
     .map(|(path, submodules)| (builder.src.join(path), submodules))
diff --git a/src/bootstrap/src/core/builder/cargo.rs b/src/bootstrap/src/core/builder/cargo.rs
index 16065cb..4e36af0 100644
--- a/src/bootstrap/src/core/builder/cargo.rs
+++ b/src/bootstrap/src/core/builder/cargo.rs
@@ -1326,10 +1326,7 @@ impl Builder<'_> {
             }
         }
 
-        if self.config.locked_deps {
-            cargo.arg("--locked");
-        }
-        if self.config.vendor || self.is_sudo {
+        if self.is_sudo {
             cargo.arg("--frozen");
         }
 
diff --git a/src/bootstrap/src/core/builder/mod.rs b/src/bootstrap/src/core/builder/mod.rs
index ae91b20..53af7f3 100644
--- a/src/bootstrap/src/core/builder/mod.rs
+++ b/src/bootstrap/src/core/builder/mod.rs
@@ -760,8 +760,6 @@ impl<'a> Builder<'a> {
                 tool::Linkchecker,
                 tool::CargoTest,
                 tool::Compiletest,
-                tool::RemoteTestServer,
-                tool::RemoteTestClient,
                 tool::RustInstaller,
                 tool::FeaturesStatusDump,
                 tool::Cargo,
@@ -770,17 +768,8 @@ impl<'a> Builder<'a> {
                 tool::Rustdoc,
                 tool::Clippy,
                 tool::CargoClippy,
-                llvm::Llvm,
-                gcc::Gcc,
-                llvm::Sanitizers,
                 tool::Rustfmt,
                 tool::Cargofmt,
-                tool::Miri,
-                tool::CargoMiri,
-                llvm::Lld,
-                llvm::Enzyme,
-                llvm::CrtBeginEnd,
-                tool::RustdocGUITest,
                 tool::OptimizedDist,
                 tool::CoverageDump,
                 tool::LlvmBitcodeLinker,
@@ -823,9 +812,6 @@ impl<'a> Builder<'a> {
                 check::CraneliftCodegenBackend,
                 check::GccCodegenBackend,
                 check::Clippy,
-                check::Miri,
-                check::CargoMiri,
-                check::MiroptTestTools,
                 check::Rustfmt,
                 check::RustAnalyzer,
                 check::TestFloatParse,
@@ -871,7 +857,6 @@ impl<'a> Builder<'a> {
                 test::CrateRustdoc,
                 test::CrateRustdocJsonTypes,
                 test::CrateBootstrap,
-                test::RemoteTestClientTests,
                 test::Linkcheck,
                 test::TierCheck,
                 test::Cargotest,
@@ -890,8 +875,6 @@ impl<'a> Builder<'a> {
                 test::EmbeddedBook,
                 test::EditionGuide,
                 test::Rustfmt,
-                test::Miri,
-                test::CargoMiri,
                 test::Clippy,
                 test::CompiletestTest,
                 test::CrateRunMakeSupport,
@@ -931,7 +914,6 @@ impl<'a> Builder<'a> {
                 doc::CargoBook,
                 doc::Clippy,
                 doc::ClippyBook,
-                doc::Miri,
                 doc::EmbeddedBook,
                 doc::EditionGuide,
                 doc::StyleGuide,
@@ -958,7 +940,6 @@ impl<'a> Builder<'a> {
                 dist::RustAnalyzer,
                 dist::Rustfmt,
                 dist::Clippy,
-                dist::Miri,
                 dist::LlvmTools,
                 dist::LlvmBitcodeLinker,
                 dist::RustDev,
@@ -989,7 +970,6 @@ impl<'a> Builder<'a> {
                 install::RustAnalyzer,
                 install::Rustfmt,
                 install::Clippy,
-                install::Miri,
                 install::LlvmTools,
                 install::Src,
                 install::RustcCodegenCranelift,
@@ -999,7 +979,6 @@ impl<'a> Builder<'a> {
                 run::BuildManifest,
                 run::BumpStage0,
                 run::ReplaceVersionPlaceholder,
-                run::Miri,
                 run::CollectLicenseMetadata,
                 run::GenerateCopyright,
                 run::GenerateWindowsSys,
diff --git a/src/build_helper/src/util.rs b/src/build_helper/src/util.rs
index 1bdbb75..861cd9c 100644
--- a/src/build_helper/src/util.rs
+++ b/src/build_helper/src/util.rs
@@ -61,6 +61,8 @@ pub fn try_run(cmd: &mut Command, print_cmd_on_fail: bool) -> Result<(), ()> {
 
 /// Returns the submodule paths from the `.gitmodules` file in the given directory.
 pub fn parse_gitmodules(target_dir: &Path) -> Vec<String> {
+    return vec![];
+
     let gitmodules = target_dir.join(".gitmodules");
     assert!(gitmodules.exists(), "'{}' file is missing.", gitmodules.display());
 
diff --git a/src/tools/rust-analyzer/crates/rust-analyzer/src/config.rs b/src/tools/rust-analyzer/crates/rust-analyzer/src/config.rs
index 0dda7f3..68c91b7 100644
--- a/src/tools/rust-analyzer/crates/rust-analyzer/src/config.rs
+++ b/src/tools/rust-analyzer/crates/rust-analyzer/src/config.rs
@@ -4118,7 +4118,8 @@ mod tests {
 
     use super::*;
 
-    #[test]
+    // Debian editors/ is pruned..
+    #[allow(dead_code)]
     fn generate_package_json_config() {
         let s = Config::json_schema();
 
